From 0a0a60f6f02dd13f526fb3ca8a5394cdc65ea069 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sun, 18 Jul 2021 01:47:44 +0200
Subject: [PATCH] Install UDev rules to the proper location by default

The comment in the CMakeLists.txt talks about some proper location the
file has to be installed too, but itself installs it to a wrong
directory. Rather than telling people to turn the option off and install
the file to the right directory manually, install it directly to the
correct directory instead. ${CMAKE_INSTALL_LIBDIR} will expand to
whatever distros require it to be
---
 CMakeLists.txt | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b924921e41..18d1b0a0b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1375,21 +1375,17 @@ if(UNIX AND NOT APPLE)
       FILES
         "${CMAKE_CURRENT_SOURCE_DIR}/res/linux/mixxx-usb-uaccess.rules"
       DESTINATION
-        "${MIXXX_INSTALL_DATADIR}/udev/rules.d"
+        "${CMAKE_INSTALL_LIBDIR}/udev/rules.d"
     )
     install(CODE "
     message(STATUS \"Important Note: Installation of udev rules\n\"
         \"The udev rule file for USB HID and Bulk controller permissions will be\n\"
         \"installed to:\n\"
-        \"    ${CMAKE_INSTALL_PREFIX}/${MIXXX_INSTALL_DATADIR}/udev/rules.d.\n\"
+        \"    ${CMAKE_INSTALL_LIBDIR}/udev/rules.d.\n\"
         \"If you are installing Mixxx from source for your own use, copy\n\"
         \"mixxx-usb-uaccess.rules to /etc/udev/rules.d/ and run:\n\"
         \"    udevadm control --reload-rules && udevadm trigger\n\"
-        \"as root to load the rules.\n\"
-        \"If you are building a package for a distribution, the correct directory for\n\"
-        \"system rules is either /lib/udev/rules.d (e.g. Debian, Fedora) or\n\"
-        \"/usr/lib/udev/rules.d (e.g. Arch Linux) with an appropriate priority prefix.\n\"
-        \"Adjust your package script accordingly and set -DINSTALL_USER_UDEV_RULES=OFF\")
+        \"as root to load the rules.\n\")
     ")
   endif()
 endif()
-- 
2.32.0

